runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -265,28 +269,58 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		if (hotKeyPressed && event.getOption().equals(WALK_HERE))
 		{
 			MenuEntry[] menuEntries = client.getMenuEntries();
-			menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+			if (!config.clearLocalMarkers())
```

Comment:
It's not good practice to start with the negative conditional first in an if-else flow.